### ISSUE TITLE
Types: db.dump() specifies an incorrect return type (and docs)

### ISSUE THREAD

**Issue by pgoldrbx** [miragejs/miragejs#819]
[db.dump()](https://github.com/miragejs/miragejs/blob/d81a6b74dc341d5395ee948b5e935af004aa367e/lib/db.js#L57-L73) is implemented to return an object of all collections, containing all rows.

- The comments and docs incorrectly state that this method `"Logs out the contents of the Db."`
- The [types definitions](https://github.com/miragejs/miragejs/blob/ad007be4f244cb19829f4985fd0920ff5305b748/types/index.d.ts#L426) specify the return type as `void`
- Also: the values, arrays of rows, are currently very loosely typed as [`any[]`](https://github.com/miragejs/miragejs/blob/ad007be4f244cb19829f4985fd0920ff5305b748/types/index.d.ts#L447), which is not ideal, but a separate issue.

To fix this, I think we could:
- Update the docs to say `"Returns the contents of the Db."`
- Update the return type to, I think, `Record<string, ModelDefinition[]>`
- Optionally also update the return type for `dbCollection.any()`

**Comment by pgoldrbx** [comment]
Unfortunately, it seems that the `lint:ts` script fails on the main branch currently, so existing issues will have to be addressed first.

```
Error: miragejs/types/tests/server-test.ts:67:5
ERROR: 67:5  expect  TypeScript@4.0 expected type to be:
  Collection<ModelInstance>
got:
  Collection<Instantiate<Registry<AnyModels, AnyFactories>, "asfd">>
```

**Comment by samselikoff** [comment]
Yeah sorry about that. I haven’t been actively working on this lately and I don’t have a typescript need so if you’d be willing to help fix master that might be the best way to move this forward!

**Comment by pgoldrbx** [comment]
I'm going to try and research this further when I have some time. Relates to #406 and the changes introduced to support Registry relationships. @dfreeman maybe you could offer some tips?

For a bit more detail, the expected type only fails under `typescript@4.x` where it seems to be fully expanding the type definitions and raising the error above.
